django/core/cache/backends/db.py

```diff
@@ -135,7 +140,14 @@ def _base_set(self, mode, key, value, timeout=DEFAULT_TIMEOUT):
                             current_expires = converter(current_expires, expression, connection, {})
 
                     exp = connection.ops.adapt_datetimefield_value(exp)
-                    if result and (mode == 'set' or (mode == 'add' and current_expires < now)):
+                    if mode == 'touch':
+                        if result:
+                            cursor.execute("UPDATE %s SET expires = %%s "
```

Comment:
Multiline is fine, but please use hanging indent:
```
cursor.execute(
    "UPDATE %s SET expires = %%s "
    ...
)
```
The other places could be changed later.